git-packages.md

```diff
@@ -0,0 +1,183 @@
+# Managing -git packages on Void
+
+*Contributed by @ericonr*
+
+Void maintainers seek to strike a balance between current software and a stable
+system. For this reason, official packages use tagged releases and avoid
+patching upstream behavior as much as possible, which makes usage of `-git`
+packages, in the style of the [AUR](https://aur.archlinux.org/), very much an
+anti-pattern. However, sometimes it is impossible to avoid this type of package,
+be it because the development branch of a piece of software you use has
+essential bug fixes or features, or because you want to test drive the brand new
+developments. This guide explains how to create such packages from an existing
+template, presents an alternative way of addding essential fixes through patches
+(which may even be accepted on the
+[void-packages](https://github.com/void-linux/void-packages) repository), and
+explains potential issues that you may face.
+
+If you have any doubts when using `xbps-src` or editing templates, be sure to
+take a look at the
+[README](https://github.com/void-linux/void-packages/blob/master/README.md) and
+the [Manual](https://github.com/void-linux/void-packages/blob/master/Manual.md)
+in void-packages.
+
+This guide uses the [conventions of the Void
+Handbook](https://docs.voidlinux.org/about/about-this-handbook.html).
+
+## Prepare the environment
+
+In order to prepare the environment, follow the steps detailed
+[here](./suckless.md#prepare-the-environment).
+
+## Converting a template into a -git template
+
+To start off, locate the template for the package you wish to convert in
+`srcpkgs/<package>/template`. Verify that the dependencies and build options
+listed in the template are enough to build the package, since features might
+have been added, removed or altered.
+
+### Fetching distfiles
+
+There are two ways of fetching the necessary distfiles: using a source control
+utility or a source control tarball. The first way has the advantage of applying
+the latest updates every time you build the package, but this can also be a
+disadvantage, since you won't be tracking the versions and reverting to a non
+broken version will be harder.
+
+For the source control utility method:
+
+- Add the utility, such as `git`, to `hostmakedepends`
+- Erase the relevant `distfiles` lines. Changing `checksum` might also be
+   necessary
+- Add a `do_fetch` function to the template, which downloads the source code
+   onto the `wrksrc` directory. Using an option equivalent to git's `--depth 1`
+   can speed up the download process. It is also possible to `cd` into the
+   downloaded directory and checkout a specific branch. An example is shown
+   below.
+
+The following example uses
+[git-clone(1)](https://man.voidlinux.org/git-clone.1).
+
+```
+do_fetch() {
+    git clone <project_url> ${wkrsrc} --depth 1
+    cd ${wrksrc}
+    git checkout <feature_branch>
+}
+```
```

Comment:
I had to add a text paragraph that wasn't an item in the list because otherwise `vmdfmt` folds the code block into the list using inline code. Might be a bug.

Reply:
Huh, okay.

Reply:
https://github.com/bobertlo/vmd/issues/10